pkg/controller/knativeserving/openshift/openshift.go

```diff
@@ -137,6 +144,53 @@ func ensureOpenshiftIngress(instance *servingv1alpha1.KnativeServing) error {
 	return nil
 }
 
+func installServiceMonitor(instance *servingv1alpha1.KnativeServing) error {
+	namespace := instance.GetNamespace()
+	log.Info("Installing ServiceMonitor")
+	const path = "deploy/service_monitor.yaml"
+
+	if serviceMonitorExists, err := serviceMonitorExists(namespace); err != nil {
+		return err
+	} else if !serviceMonitorExists {
+		log.Info("ServiceMonitor CRD is not installed. Skip to install ServiceMonitor")
+		return nil
+	}
+
+	// Add label openshift.io/cluster-monitoring to namespace
+	ns := &corev1.Namespace{}
+	if err := api.Get(context.TODO(), client.ObjectKey{Name: namespace}, ns); err != nil {
+		if !meta.IsNoMatchError(err) {
+			return err
+		}
+	}
+
+	const monitoringLabel = "openshift.io/cluster-monitoring"
+	ns.Labels[monitoringLabel] = "true"
+	if err := api.Update(context.TODO(), ns); err != nil {
+		log.Error(err, fmt.Sprintf("Could not add label %q to namespace %q", monitoringLabel, namespace))
+		return err
+	}
+
+	// Install ServiceMonitor
+	if manifest, err := mf.NewManifest(path, false, api); err == nil {
+		transforms := []mf.Transformer{mf.InjectOwner(instance)}
+		if len(namespace) > 0 {
+			transforms = append(transforms, mf.InjectNamespace(namespace))
+		}
+		if err = manifest.Transform(transforms...); err == nil {
+			err = manifest.ApplyAll()
+		}
```

Comment:
I'd do the usual if err != nil dance to have an unindented happy path:

```golang
manifest, err := mf.NewManifest(path, false, api)
if err != nil {
  log.Error(err, "Unable to create ServiceMonitor install manifest")
  return err
}
...
if err := manifest.Transform(transforms...); err != nil {
  log.Error(err, "Unable to apply transformations to service monitor manifest")
  return err
}
if err := manifest.ApplyAll(); err != nil {
  log.Error(err, "Unable to install ServiceMonitor")
  return err
}
return nil
```

That reads easier top to bottom without having to compute conditions in your head and follow branches.

Reply:
Yes, it is very easier to read. Updated.